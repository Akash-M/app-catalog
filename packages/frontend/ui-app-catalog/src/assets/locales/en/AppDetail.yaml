header: App Detail
backButton: Back to App List
details:
  id: ID
  version: Version
  description: Description
  author: Author
  url: URL
  readme: ReadMe
