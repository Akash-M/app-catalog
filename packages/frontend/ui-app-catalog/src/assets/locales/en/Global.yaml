appName: App Catalog
routes:
  catalogList: App Catalog
