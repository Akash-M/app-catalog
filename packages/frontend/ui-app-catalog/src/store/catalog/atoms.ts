import { atom } from 'recoil';

export const AppListState = atom<AppCatalog.Catalog[]>({
  key: 'AppList',
  default: [],
});
