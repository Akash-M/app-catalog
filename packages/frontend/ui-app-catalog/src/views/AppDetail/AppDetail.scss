.app-detail {
  > button {
    align-items: center;
    border: none;
    background: none;
    color: $ac-blue;
    cursor: pointer;
    display: flex;
    flex-direction: row;
    font-size: pxToRem(16);
    font-weight: bold;
    padding: 0;

    > img {
      margin-right: pxToRem(8);
    }
  }

  > section {
    > img {
      margin-bottom: pxToRem(32);
      width: 50px;
    }

    > .info {
      background-color: $ac-grey-2;
      border-radius: pxToRem(16);
      margin-bottom: pxToRem(32);

      > div {
        align-items: flex-start;
        border-bottom: 1px solid $ac-black-2;
        display: flex;
        flex-direction: column;
        padding: 0 1rem;

        @include tablet {
          align-items: center;
          flex-direction: row;
        }

        > h4 {
          color: $ac-black-3;
          margin-bottom: pxToRem(8);
          width: 100px;

          @include tablet {
            margin: pxToRem(16);
          }
        }

        &:last-child {
          border-bottom: none;
        }

        > p, a {
          margin: 0 0 pxToRem(8) 0;

          @include tablet {
            margin: pxToRem(16);
          }
        }
      }
    }
  }

  > .readme {
    background-color: $ac-grey-2;
    border-radius: pxToRem(16);
    cursor: not-allowed;
    padding: 1rem;
    pointer-events: none;
  }
}
