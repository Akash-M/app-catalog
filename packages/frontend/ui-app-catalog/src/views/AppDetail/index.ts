export { AppDetail as default } from '$/views/AppDetail/AppDetail';
