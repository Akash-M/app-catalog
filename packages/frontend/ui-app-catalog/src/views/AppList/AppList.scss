.app-list {
  &__tiles {
    display: flex;
    flex-flow: row wrap;

    > .ac-tile {
      margin: pxToRem(16);
    }
  }
}
