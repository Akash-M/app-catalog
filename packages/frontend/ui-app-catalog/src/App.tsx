import React, { useEffect } from 'react';
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import { useSetRecoilState } from 'recoil';

import { getApps } from 'lib-api/src/catalog';

import '$/App.scss';
import Layout from '$/containers/Layout';
import { AppListState } from '$/store/catalog/atoms';

const App = (): JSX.Element => {
  const setCatalogList = useSetRecoilState(AppListState);

  const fetchCatalogList = async () => {
    setCatalogList(await getApps());
  };

  useEffect(() => {
    void fetchCatalogList();
  }, []);

  return (
    <main>
      <Layout />

      <ToastContainer />
    </main>
  );
};

export default App;
