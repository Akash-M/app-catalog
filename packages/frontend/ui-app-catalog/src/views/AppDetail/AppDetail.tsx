import React, { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import ReactMarkdown from 'react-markdown';
import { useParams } from 'react-router-dom';

import { getAppById, getAppReadme } from 'lib-api/src/catalog';

import './AppDetail.scss';

export function AppDetail(): JSX.Element {
  const { t } = useTranslation(['Global', 'AppDetail']);
  // TODO: fix type.
  const { id } = useParams<any>();
  const [appDetail, setAppDetail] = useState<AppCatalog.Catalog>();
  const [appReadme, setAppReadme] = useState<string>('');
  // TODO: specify error types.
  const [error, setError] = useState(false);

  const fetchAppDetail = async () => {
    try {
      if (id) {
        const response = await getAppById(id);
        setAppDetail(response);
        if (response.readmeURL) {
          setAppReadme(await getAppReadme(response.readmeURL));
        }
      } else {
        setError(true);
      }
    } catch {
      setError(true);
    }
  };

  useEffect(() => {
    void fetchAppDetail();
  }, [id]);

  return (
    <article className="app-detail">
      <h2>{t('AppDetail.header')}</h2>

      {appDetail && !error && (
        <section className="app-detail__info">
          {appDetail.iconURL && (
            <img
              alt={appDetail.iconURL}
              className="icon"
              src={appDetail.iconURL}
            />
          )}

          <p className="title">{appDetail.name}</p>

          <p>ID: {appDetail.id}</p>

          <p>Version: {appDetail.version}</p>

          <p>Description: {appDetail.description}</p>

          {appDetail.author && <p>Author: {appDetail.author}</p>}

          {appDetail.url && (
            <p>
              <a href={appDetail.url} rel="noreferrer" target="_blank">
                {appDetail.url}
              </a>
            </p>
          )}

          {appDetail.readmeURL && <ReactMarkdown>{appReadme}</ReactMarkdown>}
        </section>
      )}
    </article>
  );
}
