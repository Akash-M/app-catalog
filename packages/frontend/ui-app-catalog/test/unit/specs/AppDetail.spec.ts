import { fireEvent, screen, waitFor } from '@testing-library/react';
import { setI18n } from 'react-i18next';

import { getAppById, getAppReadme, getApps } from 'lib-api/src/catalog';
import { customRenderer } from 'lib-utils/src/testing/factory';
import { I18N_MISSING_KEY, loadI18n } from 'lib-utils/src/testing/i18n';

import App from '$/App';
import AppDetail from '$/assets/locales/en/AppDetail.yaml';
import AppList from '$/assets/locales/en/AppList.yaml';
import Global from '$/assets/locales/en/Global.yaml';
import { AppListState } from '$/store/catalog/atoms';
import { appListFixtures } from '#/unit/fixtures/app-list';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn().mockReturnValue({ id: 'asdm123312asdnboqwedsa' }),
}));

jest.mock('lib-api/src/catalog', () => ({
  getAppById: jest.fn(),
  getAppReadme: jest.fn(),
  getApps: jest.fn(),
}));

const initializeState = ({ set }: any) => {
  set(AppListState, appListFixtures);
};

describe('<App />', () => {
  beforeAll(() => {
    setI18n(loadI18n('Global', { Global, AppDetail, AppList }));
  });

  beforeEach(jest.clearAllMocks);

  test('should render app detail based on url param and navigate back to app list' +
    'on clicking back button', async () => {
    (getAppById as jest.Mock).mockResolvedValueOnce(appListFixtures[0]);
    (getApps as jest.Mock).mockResolvedValueOnce(appListFixtures);
    (getAppReadme as jest.Mock).mockResolvedValueOnce('mock-read-me');
    const { container } = customRenderer(App, initializeState);
    await waitFor(() => expect(getApps as jest.Mock).toHaveBeenCalledTimes(1));
    fireEvent.click(screen.getByText('prometheus-operator-app-chart'));
    await waitFor(() => {
      expect(getAppById as jest.Mock).toHaveBeenCalledTimes(1);
      expect(getApps as jest.Mock).toHaveBeenCalledTimes(1);
      expect(getAppReadme as jest.Mock).toHaveBeenCalledTimes(1);
    });
    expect(container.firstChild!.textContent).not.toContain(I18N_MISSING_KEY);
    expect(container.firstChild).toMatchSnapshot();
    fireEvent.click(screen.getByText('AppDetail.backButton'));
    expect(screen.getByText('AppList.header')).toBeTruthy();
  });

  test('should display toast message in case of an error when fetching readme', async () => {
    (getAppById as jest.Mock).mockResolvedValueOnce(appListFixtures[0]);
    (getApps as jest.Mock).mockResolvedValueOnce(appListFixtures);
    (getAppReadme as jest.Mock).mockRejectedValueOnce('mock-error');
    customRenderer(App, initializeState);
    await waitFor(() => expect(getApps as jest.Mock).toHaveBeenCalledTimes(1));
    fireEvent.click(screen.getByText('prometheus-operator-app-chart'));
    await waitFor(() => {
      expect(getAppById as jest.Mock).toHaveBeenCalledTimes(1);
      expect(getApps as jest.Mock).toHaveBeenCalledTimes(1);
      expect(getAppReadme as jest.Mock).toHaveBeenCalledTimes(1);
    });
    await waitFor(() => {
      expect(screen.getByText('AppDetail.error.readme')).toBeTruthy();
    });
  });

  test('should display toast message in case of an error when fetching app details', async () => {
    (getAppById as jest.Mock).mockRejectedValueOnce('mock-error');
    (getApps as jest.Mock).mockResolvedValueOnce(appListFixtures);
    (getAppReadme as jest.Mock).mockResolvedValueOnce('mock-read-me');
    customRenderer(App, initializeState);
    await waitFor(() =>
      expect(getApps as jest.Mock).toHaveBeenCalledTimes(1));
    fireEvent.click(screen.getByText('prometheus-operator-app-chart'));
    await waitFor(() => {
      expect(getAppById as jest.Mock).toHaveBeenCalledTimes(1);
      expect(getApps as jest.Mock).toHaveBeenCalledTimes(1);
    });
    await waitFor(() => {
      expect(screen.getByText('AppDetail.error.appFetch')).toBeTruthy();
    });
  });
});
