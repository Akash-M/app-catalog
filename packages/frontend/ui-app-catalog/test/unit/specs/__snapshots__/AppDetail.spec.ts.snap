// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should display toast message in case of an error when fetching app details 1`] = `
<main>
  <article
    class="layout"
  >
    <nav
      class="site-header"
    >
      <a
        class=""
        href="/"
      >
        <h2>
          Global.appName
        </h2>
      </a>
      <div
        class="site-header__search css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-7-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class=" css-1s2u09g-control"
        >
          <div
            class=" css-319lph-ValueContainer"
          >
            <div
              class=" css-14el2xx-placeholder"
              id="react-select-7-placeholder"
            >
              Search
            </div>
            <div
              class=" css-6j8wv5-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-controls="react-select-7-listbox"
                aria-describedby="react-select-7-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                aria-owns="react-select-7-listbox"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class=""
                id="react-select-7-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-tlfecz-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <section
      class="layout__screens"
    >
      <article
        class="app-detail"
      >
        <button>
          <img
            alt="left-arrow"
            src="assets/images/arrow-left.svg"
          />
          AppDetail.backButton
        </button>
      </article>
    </section>
  </article>
  <div
    class="Toastify"
  />
</main>
`;

exports[`<App /> should display toast message in case of an error when fetching readme 1`] = `
<main>
  <article
    class="layout"
  >
    <nav
      class="site-header"
    >
      <a
        class=""
        href="/"
      >
        <h2>
          Global.appName
        </h2>
      </a>
      <div
        class="site-header__search css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-5-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class=" css-1s2u09g-control"
        >
          <div
            class=" css-319lph-ValueContainer"
          >
            <div
              class=" css-14el2xx-placeholder"
              id="react-select-5-placeholder"
            >
              Search
            </div>
            <div
              class=" css-6j8wv5-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-controls="react-select-5-listbox"
                aria-describedby="react-select-5-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                aria-owns="react-select-5-listbox"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class=""
                id="react-select-5-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-tlfecz-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <section
      class="layout__screens"
    >
      <article
        class="app-detail"
      >
        <button>
          <img
            alt="left-arrow"
            src="assets/images/arrow-left.svg"
          />
          AppDetail.backButton
        </button>
        <section>
          <h2>
            prometheus-operator-app-chart
          </h2>
          <img
            alt="https://raw.githubusercontent.com/prometheus/prometheus.github.io/master/assets/prometheus_logo-cb55bb5c346.png"
            class="icon"
            src="https://raw.githubusercontent.com/prometheus/prometheus.github.io/master/assets/prometheus_logo-cb55bb5c346.png"
          />
          <div
            class="info"
          >
            <div>
              <h4>
                AppDetail.details.id
              </h4>
               
              <p>
                asdm123312asdnboqwedsa
              </p>
            </div>
            <div>
              <h4>
                AppDetail.details.version
              </h4>
               
              <p>
                0.2.0
              </p>
            </div>
            <div>
              <h4>
                AppDetail.details.description
              </h4>
               
              <p>
                A Helm Chart for Prometheus Operator.
              </p>
            </div>
            <div>
              <h4>
                AppDetail.details.author
              </h4>
               
              <p>
                Giantswarm
              </p>
            </div>
            <div>
              <h4>
                AppDetail.details.url
              </h4>
              <a
                href="https://github.com/giantswarm/prometheus-operator-app"
                rel="noreferrer"
                target="_blank"
              >
                https://github.com/giantswarm/prometheus-operator-app
              </a>
            </div>
          </div>
        </section>
        <section
          class="readme"
        >
          <h4>
            AppDetail.details.readme
          </h4>
        </section>
      </article>
    </section>
  </article>
  <div
    class="Toastify"
  >
    <div
      class="Toastify__toast-container Toastify__toast-container--bottom-right"
    >
      <div
        class="Toastify__toast Toastify__toast-theme--light Toastify__toast--error Toastify--animate Toastify__bounce-enter--bottom-right"
        id="41zgm7ppi"
      >
        <div
          class="Toastify__toast-body"
          role="alert"
        >
          <div
            class="Toastify__toast-icon Toastify--animate-icon Toastify__zoom-enter"
          >
            <svg
              fill="var(--toastify-icon-color-error)"
              height="100%"
              viewBox="0 0 24 24"
              width="100%"
            >
              <path
                d="M11.983 0a12.206 12.206 0 00-8.51 3.653A11.8 11.8 0 000 12.207 11.779 11.779 0 0011.8 24h.214A12.111 12.111 0 0024 11.791 11.766 11.766 0 0011.983 0zM10.5 16.542a1.476 1.476 0 011.449-1.53h.027a1.527 1.527 0 011.523 1.47 1.475 1.475 0 01-1.449 1.53h-.027a1.529 1.529 0 01-1.523-1.47zM11 12.5v-6a1 1 0 012 0v6a1 1 0 11-2 0z"
              />
            </svg>
          </div>
          <div>
            AppDetail.error.readme
          </div>
        </div>
        <button
          aria-label="close"
          class="Toastify__close-button Toastify__close-button--light"
          type="button"
        >
          <svg
            aria-hidden="true"
            viewBox="0 0 14 16"
          >
            <path
              d="M7.71 8.23l3.75 3.75-1.48 1.48-3.75-3.75-3.75 3.75L1 11.98l3.75-3.75L1 4.48 2.48 3l3.75 3.75L9.98 3l1.48 1.48-3.75 3.75z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        <div
          aria-hidden="false"
          aria-label="notification timer"
          class="Toastify__progress-bar Toastify__progress-bar--animated Toastify__progress-bar-theme--light Toastify__progress-bar--error"
          role="progressbar"
          style="animation-duration: 5000ms; animation-play-state: paused; opacity: 1;"
        />
      </div>
    </div>
  </div>
</main>
`;

exports[`<App /> should render app detail based on url param and navigate back to app liston clicking back button 1`] = `
<main>
  <article
    class="layout"
  >
    <nav
      class="site-header"
    >
      <a
        class=""
        href="/"
      >
        <h2>
          Global.appName
        </h2>
      </a>
      <div
        class="site-header__search css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-2-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class=" css-1s2u09g-control"
        >
          <div
            class=" css-319lph-ValueContainer"
          >
            <div
              class=" css-14el2xx-placeholder"
              id="react-select-2-placeholder"
            >
              Search
            </div>
            <div
              class=" css-6j8wv5-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-controls="react-select-2-listbox"
                aria-describedby="react-select-2-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                aria-owns="react-select-2-listbox"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class=""
                id="react-select-2-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-tlfecz-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <section
      class="layout__screens"
    >
      <article
        class="app-detail"
      >
        <button>
          <img
            alt="left-arrow"
            src="assets/images/arrow-left.svg"
          />
          AppDetail.backButton
        </button>
        <section>
          <h2>
            prometheus-operator-app-chart
          </h2>
          <img
            alt="https://raw.githubusercontent.com/prometheus/prometheus.github.io/master/assets/prometheus_logo-cb55bb5c346.png"
            class="icon"
            src="https://raw.githubusercontent.com/prometheus/prometheus.github.io/master/assets/prometheus_logo-cb55bb5c346.png"
          />
          <div
            class="info"
          >
            <div>
              <h4>
                AppDetail.details.id
              </h4>
               
              <p>
                asdm123312asdnboqwedsa
              </p>
            </div>
            <div>
              <h4>
                AppDetail.details.version
              </h4>
               
              <p>
                0.2.0
              </p>
            </div>
            <div>
              <h4>
                AppDetail.details.description
              </h4>
               
              <p>
                A Helm Chart for Prometheus Operator.
              </p>
            </div>
            <div>
              <h4>
                AppDetail.details.author
              </h4>
               
              <p>
                Giantswarm
              </p>
            </div>
            <div>
              <h4>
                AppDetail.details.url
              </h4>
              <a
                href="https://github.com/giantswarm/prometheus-operator-app"
                rel="noreferrer"
                target="_blank"
              >
                https://github.com/giantswarm/prometheus-operator-app
              </a>
            </div>
          </div>
        </section>
        <section
          class="readme"
        >
          <h4>
            AppDetail.details.readme
          </h4>
          <p>
            mock-read-me
          </p>
        </section>
      </article>
    </section>
  </article>
  <div
    class="Toastify"
  />
</main>
`;
